src/core.ts

```diff
@@ -355,36 +355,66 @@ export type TrackingConsent = {
   [purpose: string]: boolean | undefined;
 };
 
-/** Tracking purpose consent config with timestamp & auth metadata */
-export type TrackingConsentDetails = {
-  /** Tracking consent config */
-  purposes: TrackingConsent;
-  /**
-   * Was tracking consent confirmed by the user?
-   * If this is false, the consent was resolved from defaults & is not yet confirmed
-   */
-  confirmed: boolean;
-  /** Consent resolution/last-modified timestamp (ISO 8601) */
-  timestamp: string;
-  /** Has the consent been updated (including no-change confirmation) since default resolution */
-  updated?: boolean;
-  /** Whether or not the UI has been shown to the end-user (undefined in older versions of airgap.js) */
-  prompted?: boolean;
-  /** Transparency Consent (TCF) String */
-  tcf?: string;
-  /** US Privacy (USP) String */
-  usp?: string;
-  /** Global Privacy Platform (GPP) String */
-  gpp?: string;
-  /** Consent Manager View State */
-  viewState?: ViewState;
-  /** Airgap Version */
-  airgapVersion?: string;
-  /** Arbitrary metadata that customers want to be associated with consent state */
-  metadata?: unknown;
-  /** When the metadata was last updated */
-  metadataTimestamp?: string;
-};
+export const TrackingConsentDetails = t.intersection([
+  t.type({
+    /** Tracking consent config */
+    purposes: t.record(t.string, t.boolean),
```

Comment:
ah sorry i missed this, but this is not quite right. `purposes` is a required field, but the strings can point to undefined OR booleans